pages/domains/[name]/index.tsx

```diff
@@ -149,153 +142,37 @@ export function ManageNamePageContent({ name }: { name: string }) {
-                            {ensData?.texts?.["email"]}
-                          </h3>
-                        </Link>
-                      )}
-
-                      {!!ensData?.texts?.["com.github"] && (
-                        <Link
-                          target="_blank"
-                          href={`https://github.com/${ensData?.texts["com.github"]}`}
-                          className="p-2 flex gap-2 group"
-                        >
-                          <GithubIcon className="w-5 h-5 text-gray-400 group-hover:text-black transition-colors duration-200" />
-                          <h3 className="text-gray-400 group-hover:text-black transition-colors duration-300">
-                            {ensData?.texts?.["com.github"]}
-                          </h3>
-                        </Link>
-                      )}
-
-                      {ensData?.texts?.["com.twitter"] && (
-                        <Link
-                          target="_blank"
-                          href={`https://x.com/${ensData?.texts["com.twitter"]}`}
-                          className="p-2 flex gap-2 group"
-                        >
-                          <TwitterIcon className="w-5 h-5 text-gray-400 group-hover:text-black transition-colors duration-200" />
-                          <h3 className="text-gray-400 group-hover:text-black transition-colors duration-300">
-                            {ensData?.texts?.["com.twitter"]}
-                          </h3>
-                        </Link>
-                      )}
-
-                      {ensData?.texts?.["com.linkedin"] && (
-                        <Link
-                          target="_blank"
-                          href={`https://www.linkedin.com/in/${ensData?.texts["com.linkedin"]}`}
-                          className="p-2"
-                        >
-                          <LinkedInIcon className="w-5 h-5" />
-                        </Link>
-                      )}
-                    </div>
-                  </div>
-                </div>
+                <UserDomainCard
+                  name={name}
+                  avatar={ensData?.resolver.texts?.avatar}
+                  url={ensData?.resolver.texts?.url}
+                  description={ensData?.resolver.texts?.description}
+                  email={ensData?.resolver.texts?.["email"]}
+                  github={ensData?.resolver.texts?.["com.github"]}
+                  twitter={ensData?.resolver.texts["com.twitter"]}
+                  linkedIn={ensData?.resolver.texts["com.linkedin"]}
+                  onRecordsEdited={handleFetchENSDomainData}
+                />
               </Skeleton>
 
               <div className="flex-grow flex gap-11 flex-col">
-                {!!ensData?.coins.length &&
-                  ensData?.coins.some(
+                {!!ensData?.resolver.addresses.length &&
+                  ensData?.resolver.addresses.some(
```

Comment:
When you do things like this in the UI, this array searching logic is executed in **each** Ui re-render which is costly. Might be better to extract this logic to a `useEffect` that populates an internal state which is read by this part of the Ui. What do you think?

Reply:
@FrancoAguzzi Removed it